specs/AreBrowserAcceleratorKeysEnabled.md

```diff
@@ -0,0 +1,109 @@
+that correspond to browser-specific functionality, without disabling things like
+copy and paste and movement.
+
+In this document we describe the new setting. We'd appreciate your feedback.
+
+
+## Description
+When this setting is set to false, it disables all accelerator keys that access
+features specific to the browser, including but not limited to:
+- Ctrl-F and F3 for Find on Page
+- Ctrl-P for Print
+- Ctrl-R and F5 for Reload
+- Ctrl-Plus and Ctrl-Minus for zooming
+- Ctrl-Shift-C and F12 for DevTools
+- Special keys for browser functions, such as Back, Forward, and Search
+
+It does not disable accelerator keys related to movement and text editing, such
+as:
+- Home, End, Page Up, and Page Down
+- Ctrl-X, Ctrl-C, Ctrl-V
+- Ctrl-A for Select All
+- Ctrl-Z for Undo
+
+Those accelerator keys will always be enabled unless they are handled in the
+`AcceleratorKeyPressed` event.
+
+The default value for AreBrowserAcceleratorKeysEnabled is true.
+
+## Examples
+```c#
+void ToggleBrowserKeysEnabled()
+{
+    var settings = _webView.CoreWebView2.Settings;
+    settings.AreBrowserAcceleratorKeysEnabled = !settings.AreBrowserAcceleratorKeysEnabled;
+}
+```
+
+```cpp
+void SettingsComponent::ToggleBrowserKeysEnabled()
+{
+    BOOL enabled;
+    CHECK_FAILURE(_coreWebView2Settings->get_AreBrowserAcceleratorKeysEnabled(&enabled));
+    CHECK_FAILURE(_coreWebView2Settings->put_AreBrowserAcceleratorKeysEnabled(enabled ? FALSE : TRUE));
+}
+```
+
+
+## Remarks
+Some accelerator keys that don't make sense for WebViews are always disabled
+regardless of this setting.  This includes things like opening and closing tabs,
+viewing bookmarks and history, and selecting the location bar.
+
+This setting will not prevent commands from being run through the context menu.
+To disable the context menu, use `AreDefaultContextMenusEnabled`.
+
+
+## API Details
+```
+[uuid(9aab8652-d89f-408d-8b2c-1ade3ab51d6d), object, pointer_default(unique)]
+interface ICoreWebView2Settings2 : ICoreWebView2Settings {
+    /// When this setting is set to false, it disables all accelerator keys
+    /// that access features specific to the browser, including but not limited to:
+    ///  - Ctrl-F and F3 for Find on Page
+    ///  - Ctrl-P for Print
+    ///  - Ctrl-R and F5 for Reload
+    ///  - Ctrl-Plus and Ctrl-Minus for zooming
+    ///  - Ctrl-Shift-C and F12 for DevTools
+    ///  - Special keys for browser functions, such as Back, Forward, and Search
+    ///
+    /// It does not disable accelerator keys related to movement and text editing,
+    /// such as:
+    ///  - Home, End, Page Up, and Page Down
+    ///  - Ctrl-X, Ctrl-C, Ctrl-V
+    ///  - Ctrl-A for Select All
+    ///  - Ctrl-Z for Undo
+    ///
+    /// Those accelerator keys will always be enabled unless they are handled in the
+    /// `AcceleratorKeyPressed` event.
+    /// The default value for AreBrowserAcceleratorKeysEnabled is true.
+  [propget] HRESULT AreBrowserAcceleratorKeysEnabled(
```

Comment:
We're using "browser" her as a synonym for "web view", is that an obvious connection? Today in the web view API "browser" only refers to the special process.

Reply:
What we're referring to here with "browser" is functionality that is specific to the web browser, or "browser-like" functionality, to distinguish from accelerator keys that are expected to work in all apps, such as Ctrl-C.  It is a bit ambiguous, but we could not think of a better name, and I'm not sure AreWebViewAcceleratorKeysEnabled would be any clearer.  If you have any suggestions, we'd love to here them.

Reply:
Good question. We have partitioned the browser accelerator keys into 3 groups:

1. Browser accelerator keys that don't make sense in WebView (Ctrl+T opens a new tab)
2. Browser feature related accelerator keys like Ctrl+F, Ctrl+P, Ctlr+R
3. Common control related accelerator keys like Ctrl+C, Ctrl+A

We want Are[Something]AcceleratorKeysEnabled, where [something] is identifying group 2 and not groups 1 or 3. I don't think we're using browser as a synonym for webview but as distinction from common control accelerator keys in 3. But maybe using webview2 here instead of browser would help distinguish from 1 and 3?

Reply:
Yeah, 'WebView' is the obvious alternative. WebView is the thing I'm (as the developer) interacting with, and WebView is the thing that is handling the accelerator keys.

But maybe it's find to think of it that e.g. Control+F is fundamentally a "browser accelerator key", even in a WebView. Sounds OK to me.

Reply:
To me, the distinction is "Suppose I am writing an app, and it's all written in html, running inside a WebView2, with a thin wrapper host app around it. (Extremely common pattern if you have a web site that you want to turn into an app.) I want to leave the existing web experience intact (keyboard navigation, touch scrolling, typing into edit controls), but remove all the garbage that would let the user 'break out' of my app (creating tabs, search window, F5 refresh, printing, activating developer mode, viewing source...)"

Reply:
That is exactly the use case we have in mind, and why I'm deciding to stick with AreBrowserAcceleratorKeysEnabled over AreWebViewAcceleratorKeysEnabled.  It's for when you want a WebView but not a Browser.